src/averageAges.js

```diff
@@ -14,12 +14,27 @@
  *
  * @return {number}
  */
+function getPeopleByGender(people, item) {
+  return people.filter(person => person.sex
+    === item);
+}
+
+function countYearsLiving(peopleToCout) {
+  return peopleToCout
+    .map(person => person.died - person.born)
+    .reduce((prevPerson, nextPerson) => prevPerson + nextPerson)
+    / peopleToCout.length;
+}
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men
+  = century
+    ? people.filter(person =>
+      century === Math.ceil(person.died / 100) && person.sex === 'm'
+    )
+    : getPeopleByGender(people, 'm');
```

Comment:
I'd rather format it this way
```suggestion
  const men = century
    ? people.filter(({ died, sex }) => (
         century === Math.ceil(died / 100) && sex === 'm'
       )
    )
    : getPeopleByGender(people, 'm');
```

Reply:
however, there is one more approach I'd follow
```
let men = getPeopleByGender(...);

if (century) {
  men = men.filter(...); // filtering by century given
}